### ISSUE TITLE
Assertion failure in equality engine

### ISSUE THREAD

**Issue by 4tXJ7f** [CVC4/CVC4#2002]
On the incremental QF_ANIA/UltimateBuchiAutomizer/openbsd_cstpncpy-alloca_true-termination.c.i.smt2 benchmark, we get the following assertion failure (takes around 10 seconds):

```
(error "Assertion failure.
CVC4::TNode CVC4::theory::eq::EqualityEngine::getRepresentative(CVC4::TNode) const
../../../../src/theory/uf/equality_engine.cpp:523:

  hasTerm(t)
")
```

**Comment by 4tXJ7f** [comment]
Minimized (no need to run in incremental mode):

```
(set-logic QF_ANIA)
(declare-fun x () Int)
(declare-fun a () (Array Int (Array Int Int)))
(declare-fun b () (Array Int (Array Int Int)))
(declare-fun c () (Array Int (Array Int Int)))
(assert (and (= (store a x (store (select a x) 0 0)) c) (= (store (store c 0 (store (select c 0) 0 0)) 0 (store (select (store c 0 (store (select c 0) 0 0)) 0) 0 (select (select b 0) 0))) b) (not (= 0 x))))
(check-sat)
```

**Comment by 4tXJ7f** [comment]
This issue has been solved in #2012.
